components/script/dom/xmlhttprequest.rs

```diff
@@ -1111,6 +1118,24 @@ impl XMLHttpRequest {
         blob
     }
 
+    // https://xhr.spec.whatwg.org/#arraybuffer-response
+    #[allow(unsafe_code)]
+    unsafe fn arraybuffer_response(&self, cx: *mut JSContext) -> Option<NonNullJSObjectPtr> {
+        // Step 1
+        let created = self.response_arraybuffer.get();
+        if !created.is_null() {
+            return Some(NonNullJSObjectPtr::new_unchecked(created));
+        }
+
+        // Step 2
+        let bytes = self.response.borrow().to_vec();
```

Comment:
Since we immediately make a slice of it, we shouldn't need the `to_vec` call.

Reply:
Done.